File: c/tts.c
#include <stdio.h>
#include <espeak/speak_lib.h>

int main()
{
    if (espeak_Initialize(AUDIO_OUTPUT_PLAYBACK, 0, NULL, 0) < 0)
    {
        printf("Failed to initialize eSpeak.\n");
        return 1;
    }

    const char *text = "Hello, this is a simple text-to-speech example in C.";
    espeak_Synth(text, strlen(text) + 1, 0, POS_CHARACTER, 0, espeakCHARS_AUTO, NULL, NULL);
    espeak_Synchronize();

    espeak_Terminate();

    return 0;
}

// Project and Documentation by: **Raymond C. TURNER**

// Last Updated: Saturday 9th September 2023

File: python/tts.py
from gtts import gTTS
import os

# Text to be converted to speech
text = "Return Values and Input Parameters of Methods in C#. In C#, methods can have return values and input parameters that allow them to interact with other parts of your code. Some methods are designed to perform a specific task and end quietly without returning a value. These methods are referred to as 'void methods.' On the other hand, some methods are designed to compute a result and return that value upon completion. This return value typically represents the outcome of an operation and serves as the primary means for a method to communicate back to the code that calls it."

# Create a gTTS object
tts = gTTS(text)

# Save the generated speech to a file (optional)
tts.save("output.mp3")

# Play the generated speech (optional)
os.system("mpg321 output.mp3")  # You may need to install mpg321 if not already installed

# Alternatively, you can play the speech directly without saving it
# tts.save("output.mp3")
# os.system("mpg321 output.mp3")

# Clean up the temporary speech file (optional)
# os.remove("output.mp3")


# Project and Documentation by: **Raymond C. TURNER**

# Last Updated: Monday 11th September 2023
